src/components/comparison/ComparisonDefaultCell.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <div class="default-comparison-cell" :class="valueClass">
+      {{ value ? format(value) : '-' }}
+  </div>
+</template>
+
+<script setup>
+import { computed } from 'vue';
+import { formatNumber, formatPercent } from '@utils/format.js'
+
+const props = defineProps({
+    value: Number,
+    reverseColors: Boolean,
+    valueType: {
+      type: String,
+      validator: (valueType) => ["percent", "number"].includes(valueType)
+    }
+})
+
+const valueClass = computed(() => getPositiveOrNegativeClass(props.reverseColors ? -props.value : props.value))
+function getPositiveOrNegativeClass(value) {
+  if (!value) {
+      return "gray"
+  }
+  if (value < 0) {
+      return "negative"
+  }
+  return "positive"
+}
+
+function format(value) {
+  switch (props.valueType) {
+    case "percent":
+      return formatPercent(value);
+    case "number":
+      return formatNumber(value);
+    default:
+      return "";
```

Comment:
Est-ce qu'il ne faudrait pas afficher une erreur dans la cas default? Soit un console.error, soit écrire le texte "define 'valueType' property to display your data". Ça nous permettra de facilement savoir pourquoi la valeur ne s'affiche pas si c'est à cause de ça.

Reply:
@leonarf Completement d'accord, je ne savais pas si c'était déjà une pratique ici => [`7c9de6e` (#85)](https://github.com/leonarf/VCA4D/pull/85/commits/7c9de6e39c758c4cbb03087d016f074f0b1f1b5d)

Avec du typing c'est encore mieux, tu peux avoir une sorte de [UnreachableCaseError](https://developers.ceramic.network/reference/typescript/classes/_ceramicnetwork_common.UnreachableCaseError.html) dans les `default`, ce qui permet de

- Vraiment throw une erreur si on arrive la (comme moi ici)
- Mais encore mieux: Typescript va te dire si tu a oublié un `case`, car il vérifiera que l'argument de `UnreachableCaseError(valueType)` ne peux qu'etre `never` (aka le type impossible en Typescript)